sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py

```diff
@@ -653,7 +684,8 @@ def append_taskrun_params(task_name_append: str):
     task.get("taskSpec", {})['params'] = [
         param for param in task_spec.get('params', [])
         if (task_name, param.get('name')) not in inputs_tasks or
-            param.get('name').endswith("-trname")
+            param.get('name').endswith("-trname") or
+            param.get('name').endswith("-path")
```

Comment:
Using suffixes to do some magic in the compiler seems tricky. If the user creates a task with a result named e.g. `asset-path`, it will be interpreted as a path to `asset` result (which does or does not exist), is that correct?

Reply:
These functions are to remove any generated results that are pushed into the pipelineloop or generated by the custom taskspec. I think in the newer Tekton 0.37, we can specify a result type other than `string`. So we can also looking into using result type to specify these special results.